pyracf/common/invalid_segment_name_error.py

```diff
@@ -0,0 +1,18 @@
+"""Exception to use when the user passes invalid segment name(s) on an extract request."""
+
+
+class InvalidSegmentNameError(Exception):
+    """
+    Raised when a user passes an invalid segment name on an extract request.
+    """
+
+    def __init__(self, invalid_segments: list) -> None:
+        self.message = "Building of Security Request failed.\n\n"
+        for segment in invalid_segments:
+            self.message += (
+                "Could not find "
+                + f"'{segment}' in valid segments for the requested operation.\n"
```

Comment:
I think this exception message could be made more helpful by including the type of profile in the message using the `SecurityAdmin.__profile_type` class attribute:

```python
            self.message += (
                f"'{segment}' in is not a valid segments for {self.__profile_type}.\n"
            )
```

Reply:
SegmentError is now:
`            self.message += (
                f"'{segment}' is not a valid segment for '{profile_type}'.\n"
            )`
            
SegmentTraitError is now:
`            self.message += (
                f"'{trait}' is not a valid segment-trait combination for '{profile_type}'.\n"
            )`

Corresponding changes made to unit testing as well.